src/lib/integration_test_lib/network_state.ml

```diff
@@ -30,6 +30,24 @@ module Make
           [@to_yojson
             map_to_yojson ~f:(fun ls ->
                 `List (List.map State_hash.to_yojson ls))]
+    ; blocks_seen_by_node : State_hash.Set.t String.Map.t
+          [@to_yojson
+            map_to_yojson ~f:(fun set ->
+                `List
+                  (State_hash.Set.to_list set |> List.map State_hash.to_yojson))]
+    ; blocks_including_txn : State_hash.Set.t Transaction_hash.Map.t
+          [@to_yojson
+            let open Base in
+            let set_to_yojson ~(element : 'a -> Yojson.Safe.t) s : Yojson.Safe.t
```

Comment:
Lift these up as helpers and refactor the other usages of `map_to_yojson` to use this new, more powerful definition.